src/wallet/wallet2.cpp

```diff
@@ -2118,32 +2126,6 @@ void wallet2::process_new_transaction(const crypto::hash &txid, const cryptonote
         }
       }
     }
-    else if (tx.vout.size() > 1 && tools::threadpool::getInstance().get_max_concurrency() > 1 && !is_out_data_ptr)
-    {
-      for (size_t i = 0; i < tx.vout.size(); ++i)
-      {
-        tpool.submit(&waiter, boost::bind(&wallet2::check_acc_out_precomp_once, this, std::cref(tx.vout[i]), std::cref(derivation), std::cref(additional_derivations), i,
-            std::cref(is_out_data_ptr), std::ref(tx_scan_info[i]), std::ref(output_found[i])), true);
-      }
-      THROW_WALLET_EXCEPTION_IF(!waiter.wait(), error::wallet_internal_error, "Exception in thread pool");
-
-      hw::device &hwdev = m_account.get_device();
-      boost::unique_lock<hw::device> hwdev_lock (hwdev);
-      hwdev.set_mode(hw::device::NONE);
-      for (size_t i = 0; i < tx.vout.size(); ++i)
-      {
-        THROW_WALLET_EXCEPTION_IF(tx_scan_info[i].error, error::acc_outs_lookup_error, tx, tx_pub_key, m_account.get_keys());
-        if (tx_scan_info[i].received)
-        {
-          hwdev.conceal_derivation(tx_scan_info[i].received->derivation, tx_pub_key, additional_tx_pub_keys.data, derivation, additional_derivations);
-          scan_output(tx, miner_tx, tx_pub_key, i, tx_scan_info[i], num_vouts_received, tx_money_got_in_outs, outs, pool);
-          if (!tx_scan_info[i].error)
-          {
-            tx_amounts_individual_outs[tx_scan_info[i].received->index].push_back(tx_scan_info[i].money_transfered);
-          }
-        }
-      }
-    }
```

Comment:
Unfortunately I can't figure out where these 25 lines of deleted code went, what replaced them or how they became entirely superfluous. (It's the only such deleted block that I can't place, I understood the others.)

Reply:
This deleted code would only run when the machine it's executing on has >1 thread available, otherwise it would fall through to the else, where it does the same thing but synchronously.

The function that executes in parallel in this deleted code is `check_acc_out_precomp_once` (and buried inside there is the magic function `is_out_to_acc_precomp` that view tags speed up a whole bunch). The only time `is_out_to_acc_precomp` actually executes is when it's processing new transactions that show up in the pool. So, it's safe to say that this section of the code would primarily be used to process new transactions with outputs that have view tags that show up in the pool.

I found that because checking view tags speeds up the output ownership check so drastically, that the overhead from threads outweighs the benefit you get from threading, *unless* you submit large batches into the thread pool (as is done with `geniod` on initial wallet sync -- see that comment I left close to the geniods). Since each tx in the pool has a max of 16 outputs, and most have 2, then this code in practice would actually be *slower* than just executing synchronously. So I deleted it, to fall through to the else.